adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m

```diff
@@ -0,0 +1,220 @@
+        _loadCompletionHandler(nil, error);
+        return;
+    }
+    
+    BUAdSlot *slot = [[BUAdSlot alloc] init];
+    slot.ID = placementId;
+    slot.AdType = BUAdSlotAdTypeFeed;
+    
+    _nativeAd = [[BUNativeAd alloc]initWithSlot:slot];
+    if (!_nativeAd) {
+        NSError *error = GADMAdapterPangleErrorWithCodeAndDescription(GADPangleErrorInitAd,
+                                                                      @"Init ad error, Please check the reason or you could contact Pangle"
+                                                                      );
+        _loadCompletionHandler(nil, error);
+        return;
+    }
+    _nativeAd.delegate = self;
+    [_nativeAd setAdMarkup:adConfiguration.bidResponse];
+}
+
+- (BUNativeAdRelatedView *)getRelatedView {
+    if (!_relatedView) {
+        _relatedView = [[BUNativeAdRelatedView alloc] init];
+    }
+    return _relatedView;
+}
+
+#pragma mark GADMediationNativeAd
+
+- (UIView *)mediaView {
+  if (!_mediaView) {
+    BUVideoAdView *videoView = [self getRelatedView].videoAdView;
+    _mediaView = videoView;
+  }
+  return _mediaView;
+}
+
+- (UIView *)adChoicesView {
+    return [self getRelatedView].logoADImageView;
+}
+
+- (NSString *)headline {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdTitle;
+    }
+    return nil;
+}
+
+- (NSString *)body {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdDescription;
+    }
+    return nil;
+}
+
+- (NSString *)callToAction {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.buttonText;
+    }
+    return nil;
+}
+
+- (NSDecimalNumber *)starRating {
+    return [NSDecimalNumber decimalNumberWithString:[NSString stringWithFormat:@"%ld",(long)_nativeAd.data.score]];
+}
+
+- (NSString *)store {
+    return nil;
+}
+
+- (NSString *)price {
+    return nil;
+}
+
+- (NSString *)advertiser {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdTitle;
+    }
+    return nil;
+}
+
+- (NSDictionary<NSString *,id> *)extraAssets {
+    if (_nativeAd) {
+        return @{BUDNativeAdTranslateKey:_nativeAd};
+    }
+    return nil;
+}
+
+- (void)didUntrackView:(UIView *)view {
+  [_nativeAd unregisterView];
+}
+
+- (BOOL)hasVideoContent {
+  if (_nativeAd && _nativeAd.data &&
+      (_nativeAd.data.imageMode == BUFeedVideoAdModeImage ||
+       _nativeAd.data.imageMode == BUFeedVideoAdModePortrait ||
+       _nativeAd.data.imageMode == BUFeedADModeSquareVideo))
+  {
+    return YES;
+  }
+  return NO;
+}
+
+- (void)didRenderInView:(nonnull UIView *)view
+       clickableAssetViews:
+           (nonnull NSDictionary<GADNativeAssetIdentifier, UIView *> *)clickableAssetViews
+    nonclickableAssetViews:
+        (nonnull NSDictionary<GADNativeAssetIdentifier, UIView *> *)nonclickableAssetViews
+         viewController:(nonnull UIViewController *)viewController {
+    [_nativeAd registerContainer:view withClickableViews:clickableAssetViews.allValues];
```

Comment:
Since you didn't add any logics here, I just want to confirm if Pangle SDK contains all the logics so that we don't need to filter the view/clickable/non-clickable assets here.